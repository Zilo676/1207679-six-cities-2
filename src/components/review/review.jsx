import React from 'react';
import PropTypes from 'prop-types';

import {Rating} from '../rating/rating.jsx';

const Review = (props) => {
  const {review} = props;
  return (
    <li className="reviews__item">
      <div className="reviews__user user">
        <div className="reviews__avatar-wrapper user__avatar-wrapper">
          <img className="reviews__avatar user__avatar" src={review.user.avatarUrl} width="54" height="54" alt="Reviews avatar" />
        </div>
        <span className="reviews__user-name">
          {review.user.name}
        </span>
      </div>
      <div className="reviews__info">
        <div className="reviews__rating rating">
          <div className="reviews__stars rating__stars">
            <Rating rating={review.rating} />
          </div>
        </div>
        <p className="reviews__text">
          {review.comment}
        </p>
        <time className="reviews__time" dateTime="2019-04-24">April 2019</time>
      </div>
    </li>);
};

Review.propTypes = {
  review: PropTypes.shape({
    id: PropTypes.number.isRequired,
    user: PropTypes.shape({
      id: PropTypes.number.isRequired,
      isPro: PropTypes.bool.isRequired,
      name: PropTypes.string.isRequired,
      avatarUrl: PropTypes.string.isRequired
    }),
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
  })
};

export {Review};
