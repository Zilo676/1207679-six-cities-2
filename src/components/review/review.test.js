import React from 'react';
import renderer from 'react-test-renderer';

import {Review} from './review.jsx';

it(`Correctly rendered`, () => {
  const review = {
    id: 1,
    user: {
      id: 17,
      isPro: false,
      name: `Emely`,
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/8.jpg`
    },
    rating: 4,
    comment: `I stayed here for one night and it was an unpleasant experience.`,
    date: `2019-12-05T19:40:40.338Z`
  };
  const tree = renderer.create(<Review
    review={review}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
