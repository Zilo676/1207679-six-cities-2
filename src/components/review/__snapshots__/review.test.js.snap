// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/8.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Emely
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      I stayed here for one night and it was an unpleasant experience.
    </p>
    <time
      className="reviews__time"
      dateTime="2019-12-05"
    >
      2019-12-05
    </time>
  </div>
</li>
`;
